libtsuba/include/tsuba/RDG.h

```diff
@@ -180,15 +177,21 @@ class KATANA_EXPORT RDG {
   /// Explain to graph how it is derived from previous version
   void AddLineage(const std::string& command_line);
 
+  /// Add a new RDGTopology instance to the RDG
+  void AddTopology(tsuba::RDGTopology topo);
+
   /// Load the RDG described by the metadata in handle into memory.
   static katana::Result<RDG> Make(RDGHandle handle, const RDGLoadOptions& opts);
 
-  katana::Result<void> UnbindTopologyFileStorage();
-
   /// Inform this RDG that its topology is in storage at this location
   /// without loading it into memory. \param new_top must exist and be in
   /// the correct directory for this RDG
-  katana::Result<void> SetTopologyFile(const katana::Uri& new_top);
+  /// *This function is only for use in graph conversion, and will deprecated*
+  katana::Result<void> SetCSRTopologyFile(const katana::Uri& new_top);
+
+  /// Ask this RDG if it has a topology matching the fields in shadow
+  /// If it does, the RDG returns the topology
+  katana::Result<tsuba::RDGTopology*> GetTopology(const RDGTopology& shadow);
```

Comment:
Why return a raw pointer here?

Reply:
I'm open to suggestions for other solutions (I'm still getting my cpp sea legs, so please bare with me :) )

I originally choose to return a pointer for 3(4?) reasons:

1) RDGSlice wants consistent access to what it treats as the "main" topology, and using a reference became messy fast.
2) GraphTopology needed some way of tracking which RDGTopology it had built from. This is no longer the case and will be removed (mentioned in my TODOs)
3) the memory is completely controlled by the RDGTopologyManager, and thus the RDGCore, and needs to stick around for the lifetime of the RDGCore instance so shared_ptr and unique_ptr seemed wrong to use here (but once again I could be misunderstanding something about how to use these smart pointers)
4) when you use a non-const reference, the compiler throws a warning stating to use a pointer instead. 

I hope some of that makes some sort of sense, and provides enough context.

Reply:
1. What do you mean by messy?


4. Why is it non-const?

Reply:
re: messy: making an RDGSlice instance hold a reference to a RDGTopology means the RDGSlice instance needs to be constructed with that reference. looking again, this wouldn't be too bad to do. That said, I would prefer to remove the need for RDGSlice to hold a reference/pointer to RDGTopology. 

re: non-const: RDG::GetTopology() needs a non const RDGTopology so it can be bound/mapped. GraphTopology currently needs to be able to call Store (which is non-const) on the RDGTopology (but like I mentioned elsewhere, this usage is getting removed.) so its very possible we can restrict RDG::GetTopology to returning a const RDGTopology instead.